### PR TITLE
Flush instead of synced-flush inactive shards

### PR DESCRIPTION
**Pull request by dnhatn**
If all nodes are on 7.6, we prefer to perform a normal flush instead of synced flush when a shard becomes inactive.

Backport of #49126

### CODE REVIEW

**Comment by dnhatn** [comment]
@elasticmachine run elasticsearch-ci/1

**Review by ywelsch — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/indices/flush/SyncedFlushService.java [hunk 1]
```diff
@@ -111,8 +113,13 @@ public SyncedFlushService(IndicesService indicesService,
 
     @Override
     public void onShardInactive(final IndexShard indexShard) {
-        // we only want to call sync flush once, so only trigger it when we are on a primary
-        if (indexShard.routingEntry().primary()) {
+        // A normal flush has the same effect as a synced flush if all nodes are on 7.6 or later.
+        final boolean preferNormalFlush = StreamSupport.stream(clusterService.state().nodes().spliterator(), false)
```
[the remark this answers is not in the document]
Comment (dnhatn) [reply server/src/main/java/org/elasticsearch/indices/flush/SyncedFlushService.java]: thanks, pushed https://github.com/elastic/elasticsearch/pull/51365/commits/ab93b7fd0ce392b9712f4c2af3052865b245a1b2.

**Comment by dnhatn** [comment]
> org.elasticsearch.xpack.core.ilm.WaitForSnapshotStepTests.testHashcodeAndEquals

Opened https://github.com/elastic/elasticsearch/issues/51377.

@elasticmachine run elasticsearch-ci/2

**Pull request merged**